### PR TITLE
docs: added jackson annotation example

### LINKED ISSUE #427: Show how to generate Java data models with Jackson annotation
## What to solve

In some cases, you might want to include [Jackson annotation](https://www.baeldung.com/jackson-annotations) for basic serialization.

This can be done by including the preset `JAVA_JACKSON_PRESET`.

Use the following code in the example:
```ts
generator = new JavaGenerator({ 
  presets: [JAVA_JACKSON_PRESET]
});
```

Duplicate the folder https://github.com/asyncapi/modelina/tree/master/examples/TEMPLATE and rename it to `java-generate-jackson-annotation`. Update any relevant documentation.

<details>
<summary>Some folks want more specific tasks to complete, or use this if you get stuck - CLICK ME</summary>
<p>

This issue is specially designed for first-time contributors, that want to contribute their very first code. If you are accustomed to how to solve issues in AsyncAPI, you can skip some of the steps.

#### What to know before getting started
Before actually getting started with solving the issue, there are a few things that need to be set up:
- Make sure you familiarize yourself with our [code-of-conduct](https://github.com/asyncapi/.github/blob/master/CODE_OF_CONDUCT.md) and [contributor guidelines](https://github.com/asyncapi/.github/blob/master/CONTRIBUTING.md) 
- We work on forks of repositories and branches in forks, if you are unsure how to set this up you can follow the guide on our [git-workflow setup](https://github.com/asyncapi/.github/blob/master/git-workflow.md).

Feel free to reach out either here in the issue, or on [slack under #tooling](https://www.asyncapi.com/slack-invite) if you have any questions or are stuck, we will gladly help.

## Getting started

This is a task list, that you copy into your PR description to ensure everything is included, this list walks you through each step as to how to solve this issue. It assumes that you have already set up your fork and branch and are ready to commit code.
- [ ] Duplicate the folder `https://github.com/asyncapi/modelina/tree/master/examples/TEMPLATE` and rename it to `java-generate-jackson-annotation`. This will give you a good base to start out with.
- [ ] Adapt the documentation about the feature and ensure the example are linked correctly.
    - The docs that need to change are the following: [docs/languages/Java.md#include-jackson-annotations-for-the-class](https://github.com/asyncapi/modelina/blob/master/docs/languages/Java.md#include-jackson-annotations-for-the-class)
    - The documentation must reference an actual example that is tested and contain no code itself. It is because the docs will inevitably become desynchronized and display incorrect code.
- [ ] Adapt the [examples overview](https://github.com/asyncapi/modelina/tree/master/examples/README.md) to include your new example. 
- [ ] Now it is time to adapt the test code (in examples/java-generate-jackson-annotation/index.spec.ts) and the implementation (in examples/java-generate-jackson-annotation/index.ts).
- [ ] Make sure that the test succeeds, by standing in the new example folder and execute `npm run test`.  
- [ ] Make sure that there are no linting errors, by standing in the root folder and execute `npm run lint`. 
- [ ] Create your PR, by selecting your branch that should be merged with the master branch - https://github.com/asyncapi/modelina/compare
- [ ] It is good practice to double-check your PR before requesting a review from someone else, the same way you read an assignment before handing it in.

## Do you have any examples I can check for reference?
We have currently 4 example documentations that you can look for guidance on how to achieve the desired output:
- [How to add logging to the library](https://github.com/asyncapi/modelina/tree/master/docs/advanced.md#add-logging-to-library)
- [Generate models from AsyncAPI documents](https://github.com/asyncapi/modelina/tree/master/docs/usage.md#generate-models-from-asyncapi-documents)
- [Generate models from JSON Schema draft 7 documents](https://github.com/asyncapi/modelina/tree/master/docs/usage.md#generate-models-from-json-schema-draft-7-documents)
- [Generate an interface instead of class](https://github.com/asyncapi/modelina/docs/tree/master/languages/TypeScript.md#generate-an-interface-instead-of-class)

</p>
</details>

### PR DESCRIPTION
**Pull request by Samridhi-98**
**Description**

-  Added `examples/java-generate-jackson-annotation/`
- Modified `docs/languages/Java.md`

**Related issue(s)**

Fixes #427

### CODE REVIEW

**Comment by Samridhi-98** [comment]
@jonaslagoni please review my PR :)

File: examples/java-generate-jackson-annotation/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const spy = jest.spyOn(global.console, 'log').mockImplementation(() => { return; });
+import {generate} from './index';
+
+describe('Should be able to render TEMPLATE', () => {
```
Comment (jonaslagoni) [inline examples/java-generate-jackson-annotation/index.spec.ts:4]: You need to replace`TEMPLATE` 🙂

Comment (Samridhi-98) [reply examples/java-generate-jackson-annotation/index.spec.ts]: Oh! sorry I forgot to replace it :)

**Review by jonaslagoni — APPROVED** [review]
Awesome! 🎉

**Pull request merged**